File: uva online judge/12582 - Wedding of Sultan.py
def read_test_case():
    return [ord(i) - 65 for i in input()]

def find(sequence):
    trails = [0] * 26
    stack = [sequence[0]]
    i = 1
    while len(stack) > 0:
        if stack[-1] == sequence[i]:
            stack.pop()
        else:
            trails[stack[-1]] += 1
            trails[sequence[i]] += 1
            stack.append(sequence[i])
        i += 1
    return trails

if __name__ == '__main__':
    t = int(input())
    for i in range(1, t + 1):
        print("Case {}".format(i))
        trails = find(read_test_case())
        for j in range(26):
            if trails[j] > 0:
                print("{} = {}".format(chr(j + 65), trails[j]))

File: uvaonlinejudge/12582 - Wedding of Sultan.c
#include <stdio.h>
#define N 26 

int sequence[N * N], trails[N], stack[N];

void read_test_case(int *n, int sequence[]){
    char c;
    *n = 0;
    while(1){
        scanf("%c", &c);
        if(c == '\n')
           break;
        sequence[*n] = (int) c - 65;
        (*n) += 1;
    } 
}

void find(int n, int sequence[], int trails[], int stack[]){
    int i, size;
    for(i = 0; i < N; i++)
        trails[i] = 0;
    stack[0] = sequence[0];
    size = 1;
    i = 1;
    while(size > 0){
        if(stack[size - 1] == sequence[i]){
            size--;
        }else{
            trails[stack[size - 1]]++;
            trails[sequence[i]]++;
            stack[size] = sequence[i];
            size++;
        }
        i ++;
    }
}

int main(){
    int i, j, t, n;
    scanf("%d\n", &t);
    for(i = 1; i <= t; i++){
        printf("Case %d\n", i);
        read_test_case(&n, sequence);
        find(n, sequence, trails, stack);
        for(j = 0; j < N; j++)
            if(trails[j] > 0)
                printf("%c = %d\n", j + 65, trails[j]);
    }
    return 0;
}
